### PR TITLE
📉Add support for cubicBezier() get​Point​AtLength(SVGPath, length), and interpolatePath()

### PR DESCRIPTION
**Pull request by wcandillon**
This PR adds a new function to transform an SVG path in a format that can be digested by Reanimated functions.
This enables us to build a `get​Point​AtLength()` function [identical to the one in the browsers](https://developer.mozilla.org/en-US/docs/Web/API/SVGPathElement/getPointAtLength).

RPReplay_Final1559507991 2019-06-02 22_42_16

demo 2019-06-03 14_49_56

### CODE REVIEW

**Comment by wcandillon** [comment]
@dsznajder Ready for review ;-)

File: src/Gesture.ts [hunk 1]
```diff
@@ -151,3 +152,18 @@ export const preserveMultiplicativeOffset = (
...
+}) =>
+  event(
+    [
+      {
+        nativeEvent: {
+          contentOffset
+        }
+      }
+    ],
+    { useNativeDriver: true }
```
Comment (dsznajder) [inline src/Gesture.ts]: I'm pretty sure that you don't need `useNativeDriver` for Reanimated event

Comment (wcandillon) [reply src/Gesture.ts]: true :)

File: src/CubicBezierLength.ts [hunk 2]
```diff
@@ -0,0 +1,780 @@
...
+// LUT for binomial coefficient arrays per curve order 'n'
+const binomialCoefficients = [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1]];
+
+// Look up what the binomial coefficient is for pair {n,k}
+const binomials = (n: number, k: number) => binomialCoefficients[n][k];
+
+/**
+ * Compute the curve derivative (hodograph) at t.
+ */
+const getDerivative = (derivative: number, t: number, vs: number[]): number => {
```
Comment (dsznajder) [inline src/CubicBezierLength.ts]: It's kinda magical for me :)
By the way, the code looks good but could we add some example on snack + inner in package later? As it is very advanced code and I'm pretty sure without it there might appear issues with questions `How` and `Why`.

**Comment by dsznajder** [comment]
@wcandillon Could we add some example on Snack and/or inside `Examples`? There might be problems with understanding this correctly and might appear questions for this as it's kinda advanced.

**Comment by wcandillon** [comment]
Thank you for the review. I plan to release a couple of YouTube videos about these new functions. Of course, our testing/documentation needs to substantially improve. Hopefully, we will incrementally get there as more and more people use this package.

**Pull request merged**